### PR TITLE
By default enable HIBP check

### PR DESCRIPTION
**Pull request by rullzer**
Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

**Comment by MorrisJobke** [comment]
@LukasReschke 🏓

**Comment by LukasReschke** [comment]
I think we can do this but we should also check for `has_internet_connection` here: https://github.com/nextcloud/password_policy/blob/37b2f4e45a110d97d62f9369564dd498a2118bcf/lib/Validator/HIBPValidator.php#L63-L76

**Comment by LukasReschke** [comment]
Actually I moved this a directly into the config check, this also makes it more obvious for admins in the UI that somehow HIBP is not enabled :)

https://github.com/nextcloud/password_policy/pull/148/commits/c735836b3dbd9e1702f5f73c08f7a2d9479aadc9

@MorrisJobke 🏓

**Review by LukasReschke — APPROVED** [review]
Reasonable for 22

**Review by MorrisJobke — APPROVED** [review]

File: lib/PasswordPolicyConfig.php [hunk 1]
```diff
@@ -161,10 +161,15 @@ public function setEnforceSpecialCharacters(bool $enforceSpecialCharacters) {
 	 * @return bool
 	 */
 	public function getEnforceHaveIBeenPwned(): bool {
+		$hasInternetConnection = $this->config->getSystemValue('has_internet_connection', true);
+		if(!$hasInternetConnection) {
```
Comment (MorrisJobke) [inline lib/PasswordPolicyConfig.php:165]:
```suggestion
		if (!$hasInternetConnection) {
```

**Pull request closed without merging**